### DISCUSSION TITLE
Why any-of predicate doesn't work ?

### DISCUSSION THREAD

**Discussion by uros-5** [tree-sitter/tree-sitter#3226] [Q&A] [answered]
Language: **tree-sitter-rust**
Query:
```scheme
(call_expression
  	[
    	(field_expression
        	(field_identifier) @method_name
        )
        (identifier) @method_name
        (#any-of? @method_name "render_jinja" "get_template")
    ]
    (arguments
      (string_literal)+ @template_name
    )
)

```
Source:
```rust
jinja.render_jinja("template");
jinja.get_template("template");
jinja.hello_world("template");
```
QueryCursor should capture only two methods here, but it captures three.

Here is [repository](https://github.com/uros-5/any-of-tree-sitter).

Of course I can update query:
```scheme
(call_expression
  	[
    	(field_expression
        	(field_identifier) @method_name
        )
        (identifier) @method_name
        (#match? @method_name "(render_jinja|get_template)")
    ]
    (arguments
      (string_literal)+ @template_name
    )
)

```
TreeSitter Playground doesn't highlight last method:

![screenshot](https://github.com/tree-sitter/tree-sitter/assets/59397844/65429a60-9cca-4cab-8692-4f2a0495ed7f)

**Comment by WillLillis** [comment, marked as answer]
I tried running this with your repo as is and saw the reported behavior. After that I also ran it with the `tree-sitter` dependency bumped up to the latest release. Here's the modified version of your `Cargo.toml` file to better illustrate what I mean:
```rust
[package]
name = "any-of-tree-sitter"
version = "0.1.0"
edition = "2021"

[build-dependencies]
cc="*"

[dependencies]
tree-sitter = "0.22.2"
tree-sitter-rust = {path = "/home/lillis/projects/tree-sitter-rust"}
```

In order for this to compile I also had to depend on a local clone of `tree-sitter-rust` (note the specified path for `tree-sitter-rust`) with the `tree-sitter` dependency bumped up in its `Cargo.toml` file. (The maintainers are aware of this, see [my old issue](https://github.com/tree-sitter/tree-sitter/pull/3069), and [the fix](https://github.com/tree-sitter/tree-sitter/pull/3069) they're working on) I wasn't able to locate the PR that fixed this, but it looks like this is working now under the latest tag. Hope this helps :)
